src/data/fetch.ts

```diff
@@ -6,11 +6,19 @@ import type { PineconeRecord, RecordId, RecordMetadata } from './types';
 import { Type } from '@sinclair/typebox';
 
 const RecordIdsArray = Type.Array(RecordIdSchema, { minItems: 1 });
+
+/** The list of record ids you would like to fetch using { @link Index.fetch } */
 export type FetchOptions = Array<RecordId>;
 
+/** The response from {@link Index.fetch } */
 export type FetchResponse<T extends RecordMetadata = RecordMetadata> = {
-  records?: { [key: string]: PineconeRecord<T> };
-  namespace?: string;
+  /** A map of fetched records, keyed by record id. */
+  records: {
+    [key: RecordId]: PineconeRecord<T>;
```

Comment:
It looks like TS `4.1`, `4.2`, and `4.3` don't support using a type alias as an index signature parameters type. 🤔 

Sort of annoying:
![Screenshot 2023-10-02 at 11 03 28 AM](https://github.com/pinecone-io/pinecone-ts-client/assets/119623786/39a613f0-9ac0-4eb1-9c2b-c53c954f8918)

Reply:
Annoying, but at least it's an easy fix to put it back how it was.

Reply:
Also: hooray for tests catching problems.